### PR TITLE
[6X backport]: Add ipv6 support for gpcheckperf

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
Add ipv6 support for gpcheckperf
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by kalensk** [comment]
Just a reminder to re-fly the production pipeline after merging to pickup the new gpcheckperf tests.

File: gpMgmt/test/behave/mgmt_utils/gpcheckperf.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    When  the user runs "gpcheckperf -h mdw -h sdw1 -d /data/gpdata/ -r ds"
+    Then  gpcheckperf should return a return code of 0
+    And   gpcheckperf should print "disk write tot bytes" to stdout
+
+  @concourse_cluster
+  Scenario: gpcheckperf runs runs sequential network test
+    Given the database is running
+    When  the user runs "gpcheckperf -h mdw -h sdw1 -d /data/gpdata/ -r n"
+    Then  gpcheckperf should return a return code of 0
+    And   gpcheckperf should print "avg = " to stdout
```
Comment (dkrieger-pvtl) [inline gpMgmt/test/behave/mgmt_utils/gpcheckperf.feature:16]: super nit: add a newline here.

**Review by dkrieger-pvtl — APPROVED** [review]
LGTM....the main code(the client/server) is identical to 7X.  One super nit which you can ignore.

File: gpAux/platform/gpnetbench/gpnetbenchServer.c [hunk 2]
```diff
@@ -59,69 +53,101 @@ main(int argc, char** argv)
 		return 1;
 	}
 
-	socketFd = socket(PF_INET, SOCK_STREAM, 0); 
+	rc = setupListen("::0", serverPort, AF_INET6);
+	if (rc != 0)
+		setupListen("0.0.0.0", serverPort, AF_INET);
```
Comment (dkrieger-pvtl) [inline gpAux/platform/gpnetbench/gpnetbenchServer.c:58]: This line should be:
```C
	if (rc != 0)
		rc = setupListen("0.0.0.0", serverPort, AF_INET);
```

File: gpAux/platform/gpnetbench/gpnetbenchClient.c [hunk 3]
```diff
@@ -129,38 +127,58 @@ main(int argc, char** argv)
 	{ 
 		fprintf(stderr, "socket call failed\n");
 		return 1;
-	}   
+	}
```
Comment (dkrieger-pvtl) [inline gpAux/platform/gpnetbench/gpnetbenchClient.c:131]: see https://github.com/greenplum-db/gpdb/pull/11350/files#r562979054...the old socket creation code needs to be removed also.

File: gpAux/platform/gpnetbench/gpnetbenchServer.c (hunk 2 above)
Comment (dkrieger-pvtl) [reply gpAux/platform/gpnetbench/gpnetbenchServer.c]: Done.

File: gpAux/platform/gpnetbench/gpnetbenchClient.c (hunk 3 above)
Comment (dkrieger-pvtl) [reply gpAux/platform/gpnetbench/gpnetbenchClient.c]: Done.

**Pull request merged**